src/qibolab/instruments/rfsoc.py

```diff
@@ -0,0 +1,501 @@
+            sweeper: Sweeper object
+            readouts_per_experiment: number of readout pulse to execute
+            average: if True returns averaged results, otherwise single shots
+
+        Returns:
+            Lists of I and Q value measured
+        """
+
+        # preparing the dictionary to send
+        server_commands = {
+            "operation_code": "execute_single_sweep",
+            "cfg": cfg,
+            "sequence": sequence,
+            "qubits": qubits,
+            "sweeper": sweeper,
+            "readouts_per_experiment": readouts_per_experiment,
+            "average": average,
+        }
+
+        # open a connection
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.connect((self.host, self.port))
+
+            msg_encoded = pickle.dumps(server_commands)
+            # first send 4 bytes with the length of the message
+            sock.send(len(msg_encoded).to_bytes(4, "big"))
+
+            sock.send(msg_encoded)
+
+            # wait till the server is sending
+            received = bytearray()
+            while True:
+                tmp = sock.recv(4096)
+                if not tmp:
+                    break
+                received.extend(tmp)
+        results = pickle.loads(received)
+        return results["i"], results["q"]
+
+    def play(
+        self,
+        qubits: List[Qubit],
+        sequence: PulseSequence,
+        relaxation_time: int = None,
+        nshots: int = None,
+        average: bool = False,
+    ) -> Dict[str, ExecutionResults]:
+        """Executes the sequence of instructions and retrieves readout results.
+           Each readout pulse generates a separate acquisition.
+           The relaxation_time and the number of shots have default values.
+
+        Args:
+            qubits (list): List of `qibolab.platforms.utils.Qubit` objects
+                           passed from the platform.
+            sequence (`qibolab.pulses.PulseSequence`). Pulse sequence to play.
+            nshots (int): Number of repetitions (shots) of the experiment.
+            relaxation_time (int): Time to wait for the qubit to relax to its
+                                   ground state between shots in ns.
+        Returns:
+            A dictionary mapping the readout pulses serial to
+            `qibolab.ExecutionResults` objects
+        """
+
+        # if new value are passed, they are updated in the config obj
+        if nshots is not None:
+            self.cfg.reps = nshots
+        if relaxation_time is not None:
+            self.cfg.repetition_duration = relaxation_time
+
+        toti, totq = self.call_executepulsesequence(self.cfg, sequence, qubits, len(sequence.ro_pulses), average)
+
+        results = {}
+        adcs = np.unique([qubits[p.qubit].feedback.ports[0][1] for p in sequence.ro_pulses])
+        for j in range(len(adcs)):
+            for i, ro_pulse in enumerate(sequence.ro_pulses):
```

Comment:
Yes I also think this will need changes for more qubits but, in principle, this is done because a single qick measurement operation can involve multiple adcs and the results from qick have a weird format.
In any case, for a single qubit this works fine and, in the draft for the multiqubit, I have added:
https://github.com/qiboteam/qibolab/blob/19132be641495fe7d3ead4fe332346c3bce780f3/src/qibolab/instruments/rfsoc.py#L1124-L1137